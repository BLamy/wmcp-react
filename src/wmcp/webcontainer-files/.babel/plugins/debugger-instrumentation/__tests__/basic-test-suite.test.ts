import { describe, it, expect } from 'vitest';

import { runSingle } from './helpers/runSingle';

describe('debuggerInstrumentation â€‘ inline describe/it block', () => {
  it('steps through add() & subtract()', () => {
    const steps = runSingle(
      /* js */ `
      function add(a, b) { 
        const total = a + b; 
        return total 
      }

      function subtract(a, b) { 
        const difference = a - b; 
        return difference 
      }

      describe('Math', () => {
        describe('add', () => {
          it('adds numbers', () => {
            const x = add(1, 1);
            if (x !== 2) {
              throw new Error();
            }
          });
        });

        describe('subtract', () => {
          it('subtracts numbers', () => {
            const y = subtract(5, 3);
            if (y !== 2) {
              throw new Error();
            }
          });
        });
      });
    `, 'math.spec.js', { suiteName: 'RootSuite' });

    expect(steps).toMatchSnapshot();
  });
});
