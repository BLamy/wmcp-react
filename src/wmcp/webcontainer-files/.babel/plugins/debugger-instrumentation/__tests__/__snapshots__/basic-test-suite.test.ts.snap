// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`debuggerInstrumentation â€‘ inline describe/it block > steps through add() & subtract() 1`] = `
{
  "/Users/brettlamy/Dev/wmcp-react/math.spec.js": {
    "Math": {
      "add": {
        "adds numbers": [
          {
            "line": 13,
            "sourceCode": "it('adds numbers', () => {",
            "vars": {},
          },
          {
            "line": 14,
            "sourceCode": "const x = add(1, 1);",
            "vars": {},
          },
          {
            "line": 2,
            "sourceCode": "const total = a + b;",
            "vars": {
              "a": 1,
              "b": 1,
            }
          },
          {
            "line": 3,
            "sourceCode": "return total",
            "vars": {
              "a": 1,
              "b": 1,
              "total": 2,
            }
          },
          {
            "line": 15,
            "sourceCode": "if (x !== 2) {",
            "vars": {
              "x": 2,
            },
          },
          {
            "line": 18,
            "sourceCode": "});",
            "vars": {
              "x": 2,
            },
          }
        ]
      },
      "subtract": {
        "subtracts numbers": [
          {
            "line": 21,
            "sourceCode": "it('subtracts numbers', () => {",
            "vars": {},
          },
          {
            "line": 22,
            "sourceCode": "const y = subtract(5, 3);",
            "vars": {},
          },
          {
            "line": 6,
            "sourceCode": "const difference = a - b;",
            "vars": {
              "a": 5,
              "b": 3,
            }
          },
          {
            "line": 7,
            "sourceCode": "return difference",
            "vars": {
              "a": 5,
              "b": 3,
              "difference": 2,
            }
          },
          {
            "line": 23,
            "sourceCode": "if (y !== 2) {",
            "vars": {
              "y": 2,
            },
          },
          {
            "line": 26,
            "sourceCode": "});",
            "vars": {
              "y": 2,
            },
          }
        ]
      }
    }
  }
}`;
